src/client/Microsoft.Identity.Client/TelemetryCore/Http/HttpTelemetryManager.cs

```diff
@@ -120,13 +117,22 @@ public string GetCurrentRequestHeader(ApiEvent eventInProgress)
             eventInProgress.TryGetValue(MsalTelemetryBlobEventNames.ApiIdConstStrKey, out string apiId);
             eventInProgress.TryGetValue(MsalTelemetryBlobEventNames.ForceRefreshId, out string forceRefresh);
             eventInProgress.TryGetValue(MsalTelemetryBlobEventNames.RegionDiscovered, out string regionDiscovered);
+            eventInProgress.TryGetValue(MsalTelemetryBlobEventNames.RegionSource, out string regionSource);
+
+            // Since regional fields will only be logged in case it is opted.
+            var regionalFields = new StringBuilder();
+            if (!string.IsNullOrEmpty(regionDiscovered))
+            {
+                regionalFields.Append(regionDiscovered);
+                regionalFields.Append(",");
+                regionalFields.Append((regionSource));
```

Comment:
is a check not needed for this?

Reply:
In case the region is discovered, the region source will be there and that is the reason I didn't add a check.